vertx-web/src/main/java/io/vertx/ext/web/impl/RouteImpl.java

```diff
@@ -232,6 +232,7 @@ synchronized boolean matches(RoutingContext context, String mountPoint, boolean
     }
     HttpServerRequest request = context.request();
     if (!methods.isEmpty() && !methods.contains(request.method())) {
+      ((RoutingContextImplBase) context).setDefaultErrorCode(405);
```

Comment:
@pmlopes That is against the [specification](https://tools.ietf.org/html/rfc7231#section-6.5.5) (my emphasis):

> The 405 (Method Not Allowed) status code indicates that the method
   received in the request-line is known by the origin server but **not
   supported by the target resource**.

Also:

> The origin server **MUST** generate an
   Allow header field in a 405 response containing a list of the target
   resource's currently supported methods.

In other words. For status code 405, the resource exists but it does not support the method. This is not a "should", this is a "must".